go/vt/vtgate/evalengine/fn_time.go

```diff
@@ -1200,6 +1205,54 @@ func (call *builtinMonthName) compile(c *compiler) (ctype, error) {
 	return ctype{Type: sqltypes.VarChar, Col: col, Flag: arg.Flag | flagNullable}, nil
 }
 
+func lastDay(dt datetime.DateTime) (datetime.Date, bool) {
+	ts := dt.Date.ToStdTime(time.Local)
+	firstDayOfNextMonth := time.Date(ts.Year(), ts.Month()+1, 1, 0, 0, 0, 0, time.Local)
```

Comment:
`time.Local` here should be the current timezone from the environment. Looks like that is wrong in more places, we should fix those all up then.

Reply:
@beingnoble03 If you want, I can also push up fixes for the other spots that use `time.Local`, or you can do it, whatever you prefer really.

Reply:
@dbussink I think I can do that. I'll push the changes.